lib/logmatcher.c

```diff
@@ -425,50 +441,30 @@ static gboolean
 log_matcher_pcre_re_match(LogMatcher *s, LogMessage *msg, gint value_handle, const gchar *value, gssize value_len)
 {
   LogMatcherPcreRe *self = (LogMatcherPcreRe *) s;
-  gint *matches;
-  gsize matches_size;
-  gint num_matches;
   gint rc;
 
   if (value_len == -1)
     value_len = strlen(value);
 
-  if (pcre_fullinfo(self->pattern, self->extra, PCRE_INFO_CAPTURECOUNT, &num_matches) < 0)
-    g_assert_not_reached();
-  if (num_matches > RE_MAX_MATCHES)
-    num_matches = RE_MAX_MATCHES;
-
-  matches_size = 3 * (num_matches + 1);
-  matches = g_alloca(matches_size * sizeof(gint));
-
   rc = pcre_exec(self->pattern, self->extra,
-                 value, value_len, 0, self->match_options, matches, matches_size);
-  if (rc < 0)
+                 value, value_len, 0, self->match_options, self->matches, self->matches_size);
+  if (rc == PCRE_ERROR_NOMATCH)
```

Comment:
Please split the if-s based on early return when possible, instead of else if.

Reply:
Done! :)